lib/services/flowdock.ts

```diff
@@ -53,45 +55,42 @@ export class FlowdockService extends MessageService implements ServiceEmitter, S
-                const room = event.sourceIds.room;
-                const thread = event.sourceIds.thread;
-                FlowdockService.session.get(
-                    `/flows/${org}/${room}/threads/${thread}/messages`,
-                    options,
-                    (err?: Error, result?: FlowdockMessage[]) => {
-                        // Massage the callback response into promises
-                        if(err) {
-                            reject(err);
-                        } else if(result) {
-                            resolve(_
+        }
+        return new Promise<string[]>((resolve, reject) => {
+            // Create an options array that'll shortlist the responses, if possible
+            const keywords = filter.source.match(/(\w+)/g);
+            const options = keywords === null ? {} : { search: keywords.join(' ') };
+            // Query the API via the SDK
+            const org = process.env.FLOWDOCK_ORGANIZATION_PARAMETERIZED_NAME;
+            const room = event.sourceIds.room;
+            const thread = event.sourceIds.thread;
+            FlowdockService.session.get(
+                `/flows/${org}/${room}/threads/${thread}/messages`,
+                options,
+                (error?: Error, result?: FlowdockMessage[]) => {
+                    // Massage the callback response into promises
+                    if (error) {
+                        reject(error);
+                    } else if (result) {
+                        resolve(_.map(result, (value: FlowdockMessage) => {
                                 // Clean the response to just the content
-                                .map(result, (value: FlowdockMessage) => {
-                                    return value.content;
-                                })
+                                return value.content;
+                            }).filter((value: string) => {
                                 // Filter the response to just matches
-                                .filter((value: string) => {
-                                    const match = value.match(filter);
-                                    return match !== null && match.length > 0;
-                                })
-                            );
-                        // This should never happen, since the callback pattern should have error or response
-                        // But this respects typing
-                        } else {
-                            reject(new Error('Something weird happened in the flowdock SDK'));
-                        }
+                                const match = value.match(filter);
+                                return match !== null && match.length > 0;
+                            })
+                        );
+                    // This should never happen, since the callback pattern should have error or response
+                    // ...but this respects typing
+                    } else {
```

Comment:
Yes there is, resolve & reject do not stop execution of the rest of the function.